src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java

```diff
@@ -91,42 +103,40 @@ String getUserData(ComputeOrder computeOrder) {
     }
 
     @VisibleForTesting
-    String doRequestInstance(ComputeOrder computeOrder, AzureUser azureCloudUser,
-                             AzureCreateVirtualMachineRef azureCreateVirtualMachineRef)
-            throws UnauthenticatedUserException, UnexpectedException, InstanceNotFoundException, InvalidParameterException {
-
-        this.azureVirtualMachineOperation.doCreateInstance(azureCreateVirtualMachineRef, azureCloudUser);
-        return AzureInstancePolicy.generateFogbowInstanceIdBy(computeOrder, azureCloudUser, this.defaultResourceGroupName);
+    String doRequestInstance(
+            ComputeOrder computeOrder, 
+            AzureUser azureUser,
+            AzureCreateVirtualMachineRef azureCreateVirtualMachineRef, 
+            VirtualMachineSize virtualMachineSize) throws FogbowException {
+
+        this.azureVirtualMachineOperation.doCreateInstance(azureCreateVirtualMachineRef, azureUser);
+        updateInstanceAllocation(computeOrder, virtualMachineSize);
+        return AzureInstancePolicy.generateFogbowInstanceId(computeOrder, azureUser, this.defaultResourceGroupName);
     }
 
     @VisibleForTesting
-    String getNetworkInterfaceId(ComputeOrder computeOrder, AzureUser azureCloudUser)
-            throws FogbowException {
-
+    String getNetworkInterfaceId(ComputeOrder computeOrder, AzureUser azureCloudUser) throws FogbowException {
         List<String> networkIds = computeOrder.getNetworkIds();
         if (networkIds.isEmpty()) {
-            return AzureIdBuilder
-                    .configure(azureCloudUser)
-                    .resourceGroupName(this.defaultResourceGroupName)
-                    .resourceName(this.defaultNetworkInterfaceName)
-                    .structure(AzureIdBuilder.NETWORK_INTERFACE_STRUCTURE)
-                    .build();
+            return AzureResourceIdBuilder.configure(AzureConstants.NETWORK_INTERFACE_STRUCTURE)
+                    .withSubscriptionId(azureCloudUser.getSubscriptionId())
+                    .withResourceGroupName(this.defaultResourceGroupName)
+                    .withResourceName(this.defaultNetworkInterfaceName)
+                    .buildResourceId();
```

Comment:
What do you think in use just "build()" as the method name? The "Azure ResourceId Builder" name already imply that It'll build a "resourceId".

Reply:
Ok, I'll rename.